infrastructure/azure-pipelines-pr.yaml

```diff
@@ -0,0 +1,86 @@
+name: 'Chatbot Prototype Pull Request Pipeline'
+
+pr:
+  - main
+
+variables:
+  pnpm_config_cache: $(Pipeline.Workspace)/.pnpm-store
+
+jobs:
+- job: 'chatbot_ui'
+  pool:
+    vmImage: ubuntu-latest
+  displayName: 'Chatbot UI Tests'
+  steps:
+    - template: templates/pnpm-steps.yaml
+
+    - script: |
+        pnpm tsc
+      displayName: 'Build Type System'
+      workingDirectory: chatbot-ui
+
+    - script: |
+        pnpm lint
+      displayName: 'Run ESLint'
+      workingDirectory: chatbot-ui
+
+    - script: |
+        pnpm test-ci
+      displayName: 'Run Tests'
+      workingDirectory: chatbot-ui
+
+    - task: PublishTestResults@2
+      displayName: 'Publish Test Results'
+      inputs:
+        testResultsFiles: ./chatbot-ui/junit.xml
+        mergeTestResults: true
+        testRunTitle: 'Chatbot UI Jest Unit Tests'
+
+    - task: PublishCodeCoverageResults@2
+      displayName: 'Collect Code Coverage'
+      inputs: 
+        summaryFileLocation: chatbot-ui/coverage/clover.xml
+        
+- job: 'data_ingestion_api'
+  pool:
+    vmImage: ubuntu-latest
+  strategy:
+    matrix:
+      Python311:
+        python.version: '3.11'
+  displayName: 'Data Ingestion API Tests'
+  steps:
+    - template: templates/python-steps.yaml
+
+    - script: |
+        pip install pytest pytest-azurepipelines pytest-cov
+        pipenv run python -m pytest --cov=data_ingestion data_ingestion_tests/ --cov-report=xml --test-run-title="Data Ingestion API Tests"
+      displayName: 'Run and Publish Tests'
+
+    - task: PublishCodeCoverageResults@2
+      displayName: 'Collect Code Coverage'
+      inputs: 
+        summaryFileLocation: ./coverage.xml
```

Comment:
I think before this we could run flake8 and black using 
`pipenv run black --check .` and `pipenv run flake8`. I think if both of those detect any differences in their checks then they will fail and should fail a pipeline task.